src/types.rs

```diff
@@ -279,34 +279,42 @@ pub trait Encoding {
     /// or a failure to return the error.
     #[stable]
     fn encode(&self, input: &str, trap: EncoderTrap) -> Result<Vec<u8>, Cow<'static, str>> {
+        let mut ret = Vec::new();
+        self.encode_to(input, trap, &mut ret).map(|_| ret)
+    }
+
+    /// Encode into a `ByteWriter`.
```

Comment:
Same, but not mandatory. (There is no practical encoding with a special encoding for two or more Unicode scalar values without exception.)